test/Microsoft.Azure.Devices.Edge.Test.TempSensor/Program.cs

```diff
@@ -110,4 +116,26 @@ protected override ValidationResult IsValid(object value, ValidationContext cont
             return ValidationResult.Success;
         }
     }
+
+    [AttributeUsage(AttributeTargets.Class)]
+    public sealed class InstallerPathIsRequiredOnWindowsInvalidOnLinux : ValidationAttribute
+    {
+        protected override ValidationResult IsValid(object value, ValidationContext context)
+        {
+            if (value is Program obj)
+            {
+                if (obj.InstallerPath == null && RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

Comment:
nit: it will be better, if this platform specific checking is done in platform specific class, e.g. WindowsEdgeDaemon.

Reply:
Nice, I'll do it.

Reply:
Actually, the logic is very specific to this program (refers to `Program` class, expects object to have certain program-specific fields), so it's difficult to move out right now. I'll be forced to address this when I add another test, so I'll defer until then.